packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -116,9 +143,16 @@ Future<void> buildMacOS({
     useMacOSConfig: true,
   );
 
-  // TODO(vashworth): Call `SwiftPackageManager.updateMinimumDeployment`
-  // using MACOSX_DEPLOYMENT_TARGET once https://github.com/flutter/flutter/issues/146204
-  // is fixed.
+  if (flutterProject.usesSwiftPackageManager) {
```

Comment:
This mirrors the same logic for iOS.
https://github.com/flutter/flutter/blob/978657becd854b0b5ced80d372bdfc9b82da19b0/packages/flutter_tools/lib/src/ios/mac.dart#L268-L277